frontend/www/js/omegaup/components/course/Edit.vue

```diff
@@ -216,8 +312,63 @@ import { types } from '../../api_types';
 })
 export default class CourseEdit extends Vue {
   @Prop() data!: types.CourseEditPayload;
+  @Prop() invalidParameterName!: string;
+  @Prop() initialTab!: string;
 
   T = T;
   showTab = 'course';
+
+  visibilityMode: omegaup.VisibilityMode = omegaup.VisibilityMode.Default;
+
+  assignment = emptyAssignment;
+  selectedAssignment = this.data.selectedAssignment;
+
+  get courseURL(): string {
+    return `/course/${this.data.course.alias}/`;
+  }
+
+  onNewAssignment(): void {
+    this.visibilityMode = omegaup.VisibilityMode.New;
+
+    (this.$refs.assignment_details_list as HTMLElement).scrollIntoView();
+  }
+
+  onEditAssignment(assignment: types.CourseAssignment): void {
+    this.visibilityMode = omegaup.VisibilityMode.Edit;
+
+    this.assignment = assignment;
+
+    (this.$refs.assignment_details_list as HTMLElement).scrollIntoView();
+  }
+
+  onAddProblems(assignment: types.CourseAssignment): void {
+    this.visibilityMode = omegaup.VisibilityMode.AddProblem;
+
+    this.selectedAssignment = assignment;
+    this.showTab = 'problems';
+    this.$emit('add-problems');
+  }
+
+  onCancel(): void {
+    this.$emit('cancel', this.courseURL);
+  }
+
+  onResetAssignmentForm(): void {
+    this.visibilityMode = omegaup.VisibilityMode.Default;
+  }
+
+  onSelectAssignmentTab(): void {
+    this.showTab = 'assignments';
+    this.onResetAssignmentForm();
+  }
+
+  @Watch('initialTab')
+  onInitialTabChanged(newValue: string): void {
+    if (availableTabs.includes(this.initialTab)) {
+      this.showTab = newValue;
+      return;
+    }
+    this.showTab = 'course';
```

Comment:
para usar mejor el guard clause pattern:

```suggestion
    if (!availableTabs.includes(this.initialTab)) {
      this.showTab = 'course';
      return;
    }
    this.showTab = newValue;
```

Reply:
Listo